plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsSheetTableHandle.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.google.sheets;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.connector.NotFoundException;
+
+import java.util.Objects;
+
+import static io.trino.plugin.google.sheets.SheetsClient.RANGE_SEPARATOR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class SheetsSheetTableHandle
+        implements SheetsConnectorTableHandle
+{
+    private final String sheetId;
+    private final String sheetRange;
+
+    @JsonCreator
+    public SheetsSheetTableHandle(
+            @JsonProperty("sheetId") String sheetId,
+            @JsonProperty("sheetRange") String sheetRange)
+    {
+        this.sheetId = requireNonNull(sheetId, "sheetId is null");
+        this.sheetRange = requireNonNull(sheetRange, "sheetRange is null");
+    }
+
+    @JsonProperty
+    public String getSheetId()
+    {
+        return sheetId;
+    }
+
+    @JsonProperty
+    public String getSheetRange()
+    {
+        return sheetRange;
+    }
+
+    @JsonIgnore
+    public String getSheetExpression()
+    {
+        return "%s%s%s".formatted(sheetId, RANGE_SEPARATOR, sheetRange);
+    }
+
+    @Override
+    public String toString()
+    {
+        return format("Sheet[%s]", sheetId);
```

Comment:
sheetRange?

Reply:
Good catch, originally the sheetRange was part of the sheetId but when I did the refactor to separate them I forgot to update here. Fixed.